lib/resourcebuilder/apps.go

```diff
@@ -46,6 +49,57 @@ func (b *deploymentBuilder) Do(ctx context.Context) error {
 	if b.modifier != nil {
 		b.modifier(deployment)
 	}
+
+	// if proxy injection is requested, get the proxy values and use them
+	// TODO You'll want a mutation helper that takes a pointer to a pod spec so you can wire it to other workload resources
+	if containerNamesString := deployment.Annotations["config.openshift.io/inject-proxy"]; len(containerNamesString) > 0 {
```

Comment:
Not terrible.

Reply:
Which operators would use this?  The ones that need all of their traffic to go through the proxy, including service network (I assume status.noProxy must ALWAYS contain service network or you cause an outage?)

Reply:
auth operator for example....

the noProxy will be provided by network operator which should set these (I asked him same question during arch call ;-)

Reply:
The image registry operator may use this. We already have similar logic inside of our reconciliation loop. The expectation is that noProxy covers the API server. And our operator needs to interact with external services like S3.

Reply:
> the noProxy will be provided by network operator which should set these

That's correct. I plan to enforce the default noProxy values as part of https://github.com/openshift/cluster-network-operator/pull/245

Reply:
> The expectation is that noProxy covers the API server. And our operator needs to interact with external services like S3.

@dmage team leads were updated recently that proxy support is not required if the operator only communicates in-cluster and to hosting cloud provider api's.

Reply:
> I assume status.noProxy must ALWAYS contain service network or you cause an outage?

@smarterclayton https://github.com/openshift/cluster-network-operator/pull/245 (WIP) will validate `spec.noProxy`, merge `spec.noProxy` values with the following values required by the cluster and update `status.noProxy`:

```
127.0.0.1
localhost
installConfig.ServiceNetwork
installConfig.ClusterNetwork
installConfig.MachineCIDR
apiServerURL.Hostname
internalAPIServer.Hostname
etcd-{REPL_NUM}.{CLUSTER_DOMAIN}
169.254.169.254 // cloud metadata
```

Reply:
@danehans the registry operator communicates to cloud apis even when the cluster is not running on that particular cloud.

Reply:
@bparees the registry operator is then responsible for augmenting the cluster-wide `noProxy` with additional no proxy values, correct? [This](https://github.com/openshift/cluster-network-operator/pull/245/files#diff-3ab954bacdc2df09a1ece5c1bd61c752R58-R60) is the WIP implementation of the cloud metadata ip.

Reply:
@danehans I don't know if that is desirable, particularly if a customer wants their baremetal cluster to have all outbound traffic go through their proxy (and yet still use s3 or other cloud provider storage as their backend).

Reply:
> @bparees the registry operator is then responsible for augmenting the cluster-wide noProxy with additional no proxy values, correct? 

?  My point is that the registry operator has a need to use the proxy configuration when it is talking to the cloudapis that exist outside the cloudprovider the cluster is running on.  NoProxy is irrelevant.

@dmage was observing that the registry operator has a use for this proxy env injection.  You said they didn't need it because they are talking to cloud apis, but that is incorrect.  They are talking to cloudapis, but not cloudapis from the cluster's cloud provider, therefore they will need to go through the proxy (assuming the customer's network topology requires it, which would be our assumption since it's external traffic like any other).

If the customer does not want their AWS api calls going through the proxy (and assuming their network topology doesn't require it), the admin would have to add the AWS api endpoints to their NoProxy value.

Reply:
> I don't know if that is desirable, particularly if a customer wants their baremetal cluster to have all outbound traffic go through their proxy (and yet still use s3 or other cloud provider storage as their backend).

@adambkaplan the cloud metadata ip will not be set for vsphere, baremetal and non platform types.